src/index.ts

```diff
@@ -74,6 +97,20 @@ export function normalizeLicenseText(
   return normalizedLines.join('\n').trim();
 }
 
+function stripControlCharacters(line: string, characters: string[]): string {
+  for (let i = 0; i < line.length; i++) {
+    if (characters.includes(line.charAt(i))) {
+      const lineBeforeChar = line.slice(0, i);
+      const isLastChar = i === line.length - 1;
+      const lineAfterChar = isLastChar
+        ? ''
+        : stripControlCharacters(line.slice(i + 1), characters);
+      return lineBeforeChar + lineAfterChar;
+    }
+  }
+  return line;
+}
+
```

Comment:
Went with the recursive approach as it was clean and simple. If we're concerned with blowing the stack I could make it iterative. Could also make it TCO-compatible.

Reply:
If it's run on real text files, I think we are safe. If it gets run on binary files you might blow the stack.

Reply:
Same concern. Not sure if that's a likely scenario, but I think I'd tend towards erring on the safe side given we don't know how this might be used outside of our current use case(s).

Reply:
Ok, updated to iterative approach.